src/main/kotlin/app/revanced/patches/youtube/utils/settings/VisualPreferencesIconsPatch.kt

```diff
@@ -0,0 +1,380 @@
+                "revanced_preference_screen_comments" -> "revanced_hide_quick_actions_comment_button_icon"
+                "revanced_preference_screen_feed_flyout_menu" -> "revanced_preference_screen_player_flyout_menu_icon"
+                "revanced_preference_screen_haptic_feedback" -> "revanced_enable_swipe_haptic_feedback_icon"
+                "revanced_preference_screen_patch_information" -> "about_key_icon"
+                "revanced_preference_screen_shorts_player" -> "revanced_preference_screen_shorts_icon"
+                "revanced_preference_screen_video_filter" -> "revanced_preference_screen_video_icon"
+                "revanced_swipe_gestures_lock_mode" -> "revanced_hide_player_flyout_menu_lock_screen_icon"
+                "revanced_disable_hdr_auto_brightness" -> "revanced_disable_hdr_video_icon"
+                else -> "${title}_icon"
+            }
+        }
+
+        val validExtendedBrandIcon = validExtendedBrand.associateWith { "${it}_icon" }
+
+        // Copy resources
+        val emptyIcon = "empty_icon"
+        val resourcesToCopy = mutableListOf(
+            ResourceGroup("drawable-xxhdpi", "$emptyIcon.png"),
+            ResourceGroup("drawable", *validTitlesIcons.values.map { "$it.xml" }.toTypedArray())
+        )
+
+        fun copyResourcesWithFallback(iconPath: String) {
+            try {
+                context.copyResources(
+                    iconPath,
+                    ResourceGroup("drawable", "revanced_extended_settings_key_icon.xml")
+                )
+            } catch (_: Exception) {
+                // Ignore if resource copy fails
+
+                // Add a fallback extended icon
+                // It's needed if someone provides custom path to icon(s) folder
+                // but custom branding icons for Extended setting are predefined,
+                // so it won't copy custom branding icon
+                // and will raise an error without fallback icon
+                context.copyResources(
+                    "youtube/settings/icons/extension",
+                    ResourceGroup("drawable", "revanced_extended_settings_key_icon.xml")
+                )
+            }
+        }
+
+        ExtendedIcon?.let { iconType ->
+            val selectedIconType = iconType.lowercase().replace(" ", "_")
+            CustomBrandingIconPatch.AppIcon?.let { appIcon ->
+                val appIconValue = appIcon.lowercase().replace(" ", "_")
+                val resourcePath = "youtube/branding/$appIconValue"
+
+                val iconPath = when {
+                    selectedIconType == "custom_branding_icon" -> "$resourcePath/launcher"
+                    else -> "youtube/settings/icons/$selectedIconType"
+                }
+
+                copyResourcesWithFallback(iconPath)
+            }
+        }
+
+        resourcesToCopy.forEach { context.copyResources("youtube/settings/icons", it) }
+
+        // Edit Preferences / add icon attribute
+        val tagNames = listOf(
```

Comment:
Since `android:key` is a unique value, instead of using getElementsByTagName, you can use `Node.doRecursively`